File: src/main/java/com/example/base/BaseExtension.java
package com.example.base;

import io.github.tritium_launcher.api.extension.Extension;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.koin.core.module.Module;
import java.util.Collections;
import java.util.List;

public class BaseExtension implements Extension {

    @NotNull
    @Override
    public String getNamespace() {
        return "base-extension";
    }

    @NotNull
    @Override
    public List<Module> getModules() {
        return Collections.emptyList();
    }

    @Override
    public boolean isBuiltin() {
        return false;
    }

    @Override
    public boolean getRequiresRestart() {
        return true;
    }

    @NotNull
    @Override
    public String getDisplayName() {
        return "Base Extension";
    }

    @Nullable
    @Override
    public String getDescription() {
        return "Template description";
    }
}

File: src/main/kotlin/com/example/base/BaseExtension.kt
package com.example.base

import io.github.tritium_launcher.api.extension.Extension
import org.koin.core.module.Module

class BaseExtension : Extension {
    override val namespace: String = "base-extension"
    override val displayName: String = "Base Extension"
    override val description: String = "Template description"
    override val modules: List<Module> = emptyList()
}
